CMakeLists.txt

```diff
@@ -363,6 +364,19 @@ else()
     set(openPMD_HAVE_PYTHON FALSE)
 endif()
 
+# Region algebra
+if(openPMD_USE_REGIONS STREQUAL AUTO)
+    # Enable regions by default only when building with at least C++17
+    if ("cxx_std_17" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+        set(openPMD_HAVE_REGIONS TRUE)
+    else()
+        set(openPMD_HAVE_REGIONS FALSE)
+    endif()
+elseif(openPMD_USE_REGIONS)
+    set(openPMD_HAVE_REGIONS TRUE)
```

Comment:
Sure.

Note: I find that cmake's statement "C++17 is available" is not reliable. According to cmake's documentation, cmake is really stating only "Some parts of C++17 might be available". In many cases, things don't work anyway, and I have to explicitly disable Regions in the openPMD CI builds.

Reply:
Yep, as written below I also noticed a compiler bug already. But as long as the error will in doubt not be thrown, this little extra security won't hurt too much.

On the other hand, if you and me both already observed that we would have to deactivate the region calculus in some situations manually, maybe it would be better to switch it off by default (until an eventual general switch to C++17)? A standard Cmake configuration of openPMD should try to always compile correctly out of the box, so users who just want to build and install openPMD have an easy life :)

Reply:
Done.